frontend/src/pages/pipelines/global/runs/ManageRunsTable.tsx

```diff
@@ -0,0 +1,173 @@
+  filterProps: FilterProps;
+  selectedRunIds: string[];
+  experiment: ExperimentKFv2 | null;
+};
+
+export const ManageRunsTable: React.FC<ManageRunsTableProps> = ({
+  runs,
+  experiment,
+  selectedRunIds,
+  loading,
+  totalSize,
+  page,
+  pageSize,
+  setPage,
+  setPageSize,
+  filterProps,
+  ...tableProps
+}) => {
+  const navigate = useNavigate();
+  const [isSelectAllChecked, setIsSelectAllChecked] = React.useState(false);
+  const { namespace } = usePipelinesAPI();
+  const pageRunIds = runs.map(({ run_id: runId }) => runId);
+
+  const { tableProps: checkboxTableProps, toggleSelection } = useCheckboxTable(pageRunIds);
+  const selectedIdsRef = React.useRef<string[]>(selectedRunIds);
+  const experimentId = experiment?.experiment_id ?? '';
+
+  const rowRenderer = React.useCallback(
+    (run: PipelineRunKFv2) => {
+      const isChecked = selectedIdsRef.current.includes(run.run_id);
+      const isDisabled = !isChecked && selectedIdsRef.current.length === 10;
+
+      return (
+        <PipelineRunTableRow
+          key={run.run_id}
+          checkboxProps={{
+            isChecked,
+            onToggle: () => {
+              toggleSelection(run.run_id);
+
+              // Update selected IDs for reference maintained between page changes
+              // which re - renders the table due to server - side pagination.
+              if (isChecked) {
+                selectedIdsRef.current.splice(selectedIdsRef.current.indexOf(run.run_id), 1);
+              } else {
+                selectedIdsRef.current.push(run.run_id);
+              }
+            },
+            isDisabled: !isChecked && selectedIdsRef.current.length === 10,
+            ...(isDisabled && { tooltip: 'Can only compare up to 10 runs' }),
+          }}
+          hasRowActions={false}
+          run={run}
+        />
+      );
+    },
+    [toggleSelection],
+  );
+
+  const onSelectAll = React.useCallback(
+    (isChecked: boolean) => {
+      if (!isSelectAllChecked) {
+        selectedIdsRef.current = pageRunIds;
+      } else {
+        pageRunIds.forEach((pageRunId) => {
+          if (selectedIdsRef.current.includes(pageRunId)) {
+            selectedIdsRef.current.splice(selectedIdsRef.current.indexOf(pageRunId), 1);
+          }
+        });
+      }
+      setIsSelectAllChecked(isChecked);
+    },
+    [isSelectAllChecked, pageRunIds],
+  );
+
+  return (
+    <Flex justifyContent={{ default: 'justifyContentCenter' }} data-testid="manage-runs-table">
+      <TableBase
+        {...checkboxTableProps}
+        loading={loading}
+        page={page}
+        perPage={pageSize}
+        onSetPage={(_, newPage) => {
+          if (newPage < page || !loading) {
+            setPage(newPage);
+
+            if (isSelectAllChecked) {
+              setIsSelectAllChecked(false);
+            }
+          }
+        }}
+        selectAll={{
+          onSelect: onSelectAll,
+          selected: isSelectAllChecked,
+        }}
```

Comment:
this overrides checkboxTableProps.selectAll. is this on purpose?

Reply:
Yes, for the same reason to your comment here: https://github.com/opendatahub-io/odh-dashboard/pull/2655#discussion_r1548285555

Reply:
moving convo there then